packages/shopify/api-client/src/api/category/fetchByCustomQuery.ts

```diff
@@ -0,0 +1,258 @@
+import { _shopifyCustomClient } from '../../index';
+import { ProductSearchParams, ProductsByCollectionSearchParams, Category } from '../../types';
+
+const defaultErrors = [{ message: 'an unknown error has occurred.' }];
+
+const defaultResolver = (path) => {
+  const keys = path.split('.');
+  return (_ref) => {
+    const model = _ref.model;
+    const errors = _ref.errors;
+
+    return new Promise((resolve, reject) => {
+      try {
+        const result = keys.reduce((ref, key) => {
+          return ref[key];
+        }, model);
+
+        resolve(result);
+      } catch (_) {
+        if (errors) {
+          reject(errors);
+        } else {
+          reject(defaultErrors);
+        }
+      }
+    });
+  };
+};
+
+/**
+ * Fetch collection by handle with products on the shop.
+ *
+ * @example
+ * collectionByHandleQuery(options).then((collection) => {
+ *   // Do something with the collection
+ * });
+ *
+ * @param {collectionByHandleQuery} options contains { handle, products filter query data: first, last, after, etc.. }
+ * @return {collectionByHandleQuery} A Query for collection by handle.
+ */
+const collectionByHandleQuery = (client) => {
+  const document = client.document();
```

Comment:
isnt this var name reeeealy misleading? mayeb `clientDocument` would be safer :)

Reply:
Changed the constant name to `clientDocument`